packages/clay-list/src/ListWithItems.tsx

```diff
@@ -0,0 +1,257 @@
+}
+
+interface IBooleanMap {
+	[s: string]: boolean;
+}
+
+interface IProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Property of item that makes it unique from other items.
+	 * Defaults to 'id'.
+	 */
+	itemIdentifier?: string | 'id';
+
+	/**
+	 * Items to show in list.
+	 */
+	items?: Array<IListItem>;
+
+	/**
+	 * Callback for when selected items change.
+	 */
+	onSelectedItemsChange?: (map: IBooleanMap) => void;
+
+	/**
+	 * Map of items that are currently selected.
+	 */
+	selectedItemsMap?: IBooleanMap;
+
+	/**
+	 * Path to spritemap for icon symbol.
+	 */
+	spritemap?: string;
+}
+
+const ListItem: React.FunctionComponent<
+	IListItem & {
+		selected?: boolean;
+		onSelectChange?: any;
+		spritemap?: string;
+	}
+> = ({
+	description,
+	dropdownActions,
+	href,
+	labels,
+	onSelectChange,
+	quickActions,
+	selected = false,
+	spritemap,
+	title,
+}) => {
+	return (
+		<ClayList.Item active={selected} flex>
+			{onSelectChange && (
+				<ClayList.ItemField>
+					<ClayCheckbox
+						aria-label="checkbox"
+						checked={selected}
+						onChange={() => {
+							onSelectChange(!selected);
+						}}
+					/>
+				</ClayList.ItemField>
+			)}
+
+			<ClayList.ItemField>
+				<ClaySticker displayType="secondary">
+					<ClayIcon spritemap={spritemap} symbol="folder" />
+				</ClaySticker>
+			</ClayList.ItemField>
+
+			<ClayList.ItemField expand>
+				<ClayList.ItemTitle href={href}>{title}</ClayList.ItemTitle>
+
+				<ClayList.ItemText>{description}</ClayList.ItemText>
+
+				{labels && (
+					<div className="list-group-detail">
+						{labels.map(({value, ...others}, j) => (
+							<ClayLabel {...others} key={j}>
+								{value}
+							</ClayLabel>
+						))}
+					</div>
+				)}
+			</ClayList.ItemField>
+
+			<ClayList.ItemField>
+				{quickActions && (
+					<ClayList.QuickActionMenu>
+						{quickActions.map((action, j) => (
+							<ClayList.QuickActionMenu.Item
+								key={j}
+								onClick={action.onClick}
+								spritemap={spritemap}
+								symbol={action.symbol}
+							/>
+						))}
+					</ClayList.QuickActionMenu>
+				)}
+
+				{dropdownActions && (
+					<ClayDropDownWithBasicItems
+						items={dropdownActions}
+						spritemap={spritemap}
+						trigger={
+							<button className="component-action">
+								<ClayIcon
+									spritemap={spritemap}
+									symbol="ellipsis-v"
+								/>
+							</button>
+						}
+					/>
+				)}
+			</ClayList.ItemField>
+		</ClayList.Item>
+	);
+};
+
+export const ClayListWithItems: React.FunctionComponent<IProps> = ({
+	className,
+	itemIdentifier = 'id',
+	items = [],
+	selectedItemsMap = {},
+	onSelectedItemsChange,
+	spritemap,
+	...otherProps
+}) => {
+	return (
+		<div {...otherProps} className={classNames(className)}>
+			<ClayList>
+				{items.map((item, i) => {
+					if (item.items) {
+						return (
+							<React.Fragment key={`headerGroup${i}`}>
+								<ClayList.Header>{item.header}</ClayList.Header>
+
+								{item.items.map(subItem => {
```

Comment:
This code block can be separated into some other component and called recursively when there is an `items` in the props, you can get a better reuse of the code here.